### PR TITLE
Add meme_psp_gen tool

### PR DESCRIPTION
**Pull request by gregvonkuster**

### CODE REVIEW

**Comment by martenson** [comment]
@gregvonkuster this is probably targeting wrong repo

**Comment by mvdbeek** [comment]
If you want to trigger the tests you need to add a .shed.yml file

**Comment by mvdbeek** [comment]
nevermind, that's targeting the existing meme repo, so that's OK.

File: tools/meme/meme_psp_gen.xml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                <option value="yes">Yes</option>
+            </param>
+            <conditional name="report_scores_cond">
+                <param name="report_scores" type="select" label="Output primary and control file names, scores and widths?">
+                    <option value="no" selected="true">No</option>
+                    <option value="yes">yes</option>
+                </param>
+                <when value="no"/>
+                <when value="yes">
+                    <param name="verbose" type="select" display="radio" label="Report frequency of each score?">
```
Comment (bgruening) [inline tools/meme/meme_psp_gen.xml]: boolean

File: tools/meme/meme_psp_gen.xml [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                </param>
+                <when value="no"/>
+                <when value="yes">
+                    <param name="fixed_start" type="select" display="radio" label="Allow triples to start anywhere within a site?" help="Select 'No' to only consider triples starting at the start of the site">
+                        <option value="yes" selected="true">Yes</option>
+                        <option value="no">No</option>
+                    </param>
+                </when>
+            </conditional>
+            <param name="equiv" type="select" display="radio" label="Match as equal sequences of letters that appear together?">
```
Comment (bgruening) [inline tools/meme/meme_psp_gen.xml]: boolean and the ones below

File: tools/meme/meme_psp_gen.xml [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+#if str($adv.scale_min) == 'yes':
+    -scalemin
+#end if
+#if str($adv.scale_max) == 'yes':
+    -scalemax
+#end if
+#if str($adv.max_range) == 'yes':
+    -maxrange
+#end if
+#if str($adv.raw) == 'yes':
```
Comment (bgruening) [inline tools/meme/meme_psp_gen.xml]: If all these are booleans you can simply write: `$adv.raw`

File: tools/meme/meme_psp_gen.xml [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+* **Set the highest score value after scaling** - select 'Yes' to set the highest score to 0.9 unless the previous "lowest score" option is selected, in which case the highest score is lowest score + 1.
+* **Choose the width with the biggest difference between minimum and maximum scores before scaling** - select 'Yes' to choose the width with the biggest difference between minimum and maximum scores before scaling, or 'No' to choose the width with the biggest maximum score before scaling.
+* **Output scores instead of priors** - select 'Yes' to output scores instead of position specific priors.
+* **Output primary and control file names, scores and widths** - select 'Yes' to produce an additional tabular output consisting of control file names, lowest and highest scores and lowest and highest widths.
+
+   * **Report frequency of each score** - select 'Yes' to include the frequency of each score in the output.
+
+    </help>
+    <citations>
+        <citation type="bibtex">
```
Comment (bgruening) [inline tools/meme/meme_psp_gen.xml]: you could use this DOI here: `10.1186/1471-2105-11-179`.
Keep in mind that most if not all papers that are really published do have a DOI these days. Bibtex we should use if there is no classical publication available.

File: tools/meme/meme_psp_gen.xml [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                    <option value="yes">yes</option>
+                </param>
+                <when value="no"/>
+                <when value="yes">
+                    <param name="fixed_start" type="boolean" truevalue="-fixedstart" falsevalue="" checked="False" label="Allow triples to start anywhere within a site?" help="Select 'No' to only consider triples starting at the start of the site"/>
+                </when>
+            </conditional>
+            <param name="equiv" type="boolean" truevalue="-equiv" falsevalue="" checked="False" label="Match as equal sequences of letters that appear together?"/>
+            <param name="consider_both_strands" type="boolean" truevalue="-revcomp" falsevalue="" checked="False" label="Consider both strands when calculating position specific priors for alphabets?"/>
+            <param name="scale_min" type="boolean" truevalue="-scalemin" falsevalue="" checked="False" label="Set the lowest score value after scaling?"/>
```
Comment (bgruening) [inline tools/meme/meme_psp_gen.xml]: This looks like a perfect fit for `argument="-scalemin"` (https://docs.galaxyproject.org/en/latest/dev/schema.html#id14). You can remove the name tag afterwards and get a variable called `scalemin`, in addition you get a nicely formatted help text.

File: tools/meme/meme_psp_gen.xml [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+                <option value="-rna">RNA</option>
+            </param>
+            <conditional name="triples_cond">
+                <param name="triples" type="select" label="Use spaced triples instead of whole-word matches?" help="Recommended for protein">
+                    <option value="no" selected="true">No</option>
+                    <option value="yes">yes</option>
+                </param>
+                <when value="no"/>
+                <when value="yes">
+                    <param name="fixed_start" type="boolean" truevalue="-fixedstart" falsevalue="" checked="False" label="Allow triples to start anywhere within a site?" help="Select 'No' to only consider triples starting at the start of the site"/>
```
Comment (bgruening) [inline tools/meme/meme_psp_gen.xml]: argument?

File: tools/meme/meme_psp_gen.xml [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+                </when>
+            </conditional>
+            <param name="equiv" type="boolean" truevalue="-equiv" falsevalue="" checked="False" label="Match as equal sequences of letters that appear together?"/>
+            <param name="consider_both_strands" type="boolean" truevalue="-revcomp" falsevalue="" checked="False" label="Consider both strands when calculating position specific priors for alphabets?"/>
+            <param name="scale_min" type="boolean" truevalue="-scalemin" falsevalue="" checked="False" label="Set the lowest score value after scaling?"/>
+            <param name="scale_max" type="boolean" truevalue="-scalemax" falsevalue="" checked="False" label="Set the highest score value after scaling?"/>
+            <param name="max_range" type="boolean" truevalue="-maxrange" falsevalue="" checked="False" label="Choose the width with the biggest difference between minimum and maximum scores before scaling?" help="Select 'No' to choose the width with the biggest maximum score (before scaling)"/>
+            <param name="raw" type="boolean" truevalue="-raw" falsevalue="" checked="False" label="Output scores instead of priors?"/>
```
Comment (bgruening) [inline tools/meme/meme_psp_gen.xml]: all can use argument if you like

File: tools/meme/meme_psp_gen.xml [hunk 8]
```diff
@@ -0,0 +1,140 @@
+<tool id="meme_psp_gen" name="MEME psp-gen" version="4.11.2.0">
```
Comment (bgruening) [inline tools/meme/meme_psp_gen.xml]: what do you think about putting the version also into a macro?

**Comment by bgruening** [comment]
Thanks @gregvonkuster!

**Pull request merged**